src/Microsoft.OData.Core/TelemetryHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TelemetryHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData
+{
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+    using Activity = System.Diagnostics.Activity;
+    using ActivitySource = System.Diagnostics.ActivitySource;
+#else
+    using System;
+#endif
+
+    internal static class TelemetryHelper
+    {
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+        public static readonly ActivitySource ODataTelemetrySource = new ActivitySource("Microsoft.OData.Core");
+#endif
+
+        public static Activity StartActivity(string displayName)
+        {
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+            return ODataTelemetrySource.StartActivity(displayName);
+#else
+            return null;
+#endif
+        }
+
+        public static void StopActivity()
+        {
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+            Activity.Current?.Stop();
+#endif
+        }
+
+        public static void StopAndDisposeActivity()
+        {
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+            Activity.Current?.Dispose(); // Stops and disposes activity
+#endif
+        }

```

Comment:
I don't think we need both of this since `Dispose()` _just_ stops the activity. Maybe we can keep one of these methods and just call `Activity.Current?.Dispose`.